tapir/shifts/services/slot_modification_service.py

```diff
@@ -0,0 +1,194 @@
+import csv
+import datetime
+import io
+from dataclasses import dataclass
+from typing import List
+
+from django.db import transaction
+from django.db.models import Q
+from django.utils import timezone
+
+from tapir.accounts.models import TapirUser
+from tapir.shifts.models import (
+    ShiftTemplate,
+    ShiftSlotTemplate,
+    ShiftAttendanceTemplate,
+    ShiftAttendance,
+    ShiftSlot,
+)
+from tapir.utils.user_utils import UserUtils
+
+
+class SlotModificationService:
+    @dataclass(frozen=True)
+    class ParameterSet:
+        workday_or_weekend: str
```

Comment:
Why there is a need to have a special handling for workday vs. weekend?

Reply:
During the week, the shifts are (almost) always the same regardless of the day, so when we want to change for example "monday 8:15", we usually want to change also tuesday, wednesday... for all 4 ABCD weeks. 
Saturday shifts are layed out a bit differently, so the changes are usually specific to that day.
It's also described in the issue #496

Reply:
Okay, that makes sense.